sdk/metric/pipeline.go

```diff
@@ -351,8 +351,8 @@ func (i *inserter[N]) cachedAggregator(scope instrumentation.Scope, kind Instrum
 		b := aggregate.Builder[N]{
 			Temporality: i.pipeline.reader.temporality(kind),
 		}
-		if len(stream.AllowAttributeKeys) > 0 {
-			b.Filter = stream.attributeFilter()
+		if stream.AttributeFilter != nil {
+			b.Filter = stream.AttributeFilter
 		}
```

Comment:
Can this be just `b.Filter = stream.AttributeFilter`?  If it's nil it will stay nil, if not, we get a filter.